### PR TITLE
Adding a rst discovery plugin, adding rst-lint and rstcheck tool plug…

### PR DESCRIPTION
**Pull request by xydesa**
…ins.

Signed-off-by: Alexander Xydes <alexander.xydes@navy.mil>

### CODE REVIEW

**Comment by xydesa** [comment]
@tdenewiler apparently Linux mint supports rst files in it's "magic file" for the file command: /usr/share/misc/magic.mgc
but Mac OS X, and Ubuntu do not. I think the two options are either: don't test this feature, or drop support for it. I'm inclined to drop support for it as it doesn't see to be widely supported in the OS's and I don't know how to distinguish this support.
what do you think?

**Comment by tdenewiler** [comment]
I agree that dropping support for a feature that is not widely supported among OS's is a good approach.

File: src/statick_md/plugins/discovery/rst_discovery_plugin.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    def get_name(self) -> str:
+        """Get name of discovery type."""
+        return "rst"
+
+    def scan(
+        self, package: Package, level: str, exceptions: Optional[Exceptions] = None
+    ) -> None:
+        """Scan package looking for rst files."""
+        src_files = []  # type: List[str]
```
Comment (denewiler) [inline src/statick_md/plugins/discovery/rst_discovery_plugin.py:23]:
```suggestion
        src_files: List[str] = []
```

File: src/statick_md/plugins/tool/rstcheck_tool_plugin.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    # pylint: disable=too-many-locals
+    def scan(self, package: Package, level: str) -> Optional[List[Issue]]:
+        """Run tool and gather output."""
+        tool_bin = "rstcheck"
+
+        flags = []  # type: List[str]
+        user_flags = self.get_user_flags(level)
+        flags += user_flags
+
+        files = []  # type: List[str]
```
Comment (denewiler) [inline src/statick_md/plugins/tool/rstcheck_tool_plugin.py:29]:
```suggestion
        files: List[str] = []
```

File: src/statick_md/plugins/tool/rstcheck_tool_plugin.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            for line in output.split("\n"):
+                match = parse.match(line)  # type: Optional[Match[str]]
+                if match:
+                    issues.append(
+                        Issue(
+                            match.group(1),
+                            match.group(2),
+                            self.get_name(),
+                            match.group(3),
+                            int(match.group(4)),
```
Comment (denewiler) [inline src/statick_md/plugins/tool/rstcheck_tool_plugin.py:85]: The severity can be left as a string.

```suggestion
                            match.group(4),
```

**Review by denewiler — APPROVED** [review]

**Review by denewiler — APPROVED** [review]

**Pull request merged**